src/blas/highlevel.jl

```diff
@@ -10,8 +10,17 @@ CublasArray{T<:CublasFloat} = CuArray{T}
 #
 ###########
 
-Base.scale!(x::CuArray{T}, k::Number) where T<:CublasFloat =
-  scal!(length(x), convert(eltype(x), k), x, 1)
+function Base.scale!(X::CuArray{T}, s::Real) where T <: Base.BLAS.BlasComplex
```

Comment:
Why is this needed? cublas already has methods for complex numbers

Reply:
Ambiguities! But I will double check.

Reply:
Even with the ambiguities it makes sense to use CUBLAS's method for complex numbers rather than reimplementing it.

Reply:
The ambiguity here is caused by GPUArrays wrapping the method for Real rather than Number. So I have to explicitly override that in order to opt-out I guess, which is a bit nasty.

Reply:
Okay lets fix that then ;)

Reply:
https://github.com/JuliaGPU/GPUArrays.jl/pull/83